### PR TITLE
Github module

### PR DESCRIPTION
**Pull request by tobes**
Display Github notifications and issue/pull requests for a repo.

To check notifications a Github `username` and `personal access token` are
required.  You can create a personal access token at
https://github.com/settings/tokens The only `scope` needed is `notifications`,
which provides readonly access to notifications.

The Github API is rate limited so setting `cache_timeout` too small may cause
issues see https://developer.github.com/v3/#rate-limiting for details

Configuration parameters:
  - `auth_token` Github personal access token, needed to check notifications
    see above.
    *(default None)*
  - `button_notifications` Button that when clicked opens the notification page
    on github.com. Setting to `0` disables.
    *(default 1)*
  - `cache_timeout` How often we refresh this module in seconds
    *(default 60)*
  - `format` Format of output
    (default '{repo} {issues}/{pull_requests}{notifications}'
    if username and auth_token provided else
    '{repo} {issues}/{pull_requests}')
  - `format_notifications` Format of `{notification}` status placeholder.
    *(default ' 🔔{notification_count}')*
  - `notifications` Type of notifications can be `all` for all notifications or
    `repo` to only get notifications for the repo specified.  If repo is
    not provided then all notifications will be checked.
    *(default 'all')*
  - `repo` Github repo to check
    *(default 'ultrabug/py3status')*
  - `username` Github username, needed to check notifications.
    *(default None)*

Format of status string placeholders:
  - `{repo}` the name of the repository being checked.
  - `{issues}` Number of open issues.
  - `{pull_requests}` Number of open pull requests
  - `{notifications}` Notifications.  If no notifications this will be empty.
  - `{notification_count}` Number of notifications.  This is also the __Only__
    status string available to `format_notifications`.

Requires:
  - `requests` python module from pypi https://pypi.python.org/pypi/requests

Examples:

```
# set github access credentials
github {
    auth_token = '40_char_hex_access_token'
    username = 'my_username'
}

# just check for any notifications
github {
    auth_token = '40_char_hex_access_token'
    username = 'my_username'
    format = 'Github {notification_count}'
}
```

### CODE REVIEW

**Comment by tobes** [comment]
I've updated this.

* We now don't check notifications in the default format if `username` and `auth_token` are not supplied.

* `{notification_count}` status string placeholder added.

* docstring updated to reflect this.

**Comment by ultrabug** [comment]
omg, one of my oldest wishlist module getting real ? testing !

File: py3status/modules/github.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+GITHUB_API_URL = 'https://api.github.com'
+GITHUB_NOTIFICATION_URL = 'https://github.com/notifications'
+
+
+class Py3status:
+    auth_token = None
+    button_notifications = 1
+    cache_timeout = 60
+    format = None
+    format_notifications = u' 🔔{notification_count}'
```
Comment (ultrabug) [inline py3status/modules/github.py]: This unicode bell doesn't show on my "standard" font :(

File: py3status/modules/github.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    notifications = 'all'
+    repo = 'ultrabug/py3status'
```
Comment (ultrabug) [inline py3status/modules/github.py]: Maybe a `short_repo_name` option defaulting to True would be  cool to save space on the bar ? and we would see py3status displayed only

File: py3status/modules/github.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        self.repo_warning = False
+        self._issues = '?'
+        self._pulls = '?'
+
+    def _init(self):
+        # Set format if user has not configured it.
+        if not self.format:
+            if self.username and self.auth_token:
+                # include notifications
+                self.format = '{repo} {issues}/{pull_requests}{notifications}'
```
Comment (ultrabug) [inline py3status/modules/github.py]: Something with the unicode char on notifications but it breaks on python2.7

File: py3status/modules/github.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+github {
+    auth_token = '40_char_hex_access_token'
+    username = 'my_username'
+}
+
+# just check for any notifications
+github {
+    auth_token = '40_char_hex_access_token'
+    username = 'my_username'
+    format = 'Github {notification_count}'
```
Comment (ultrabug) [inline py3status/modules/github.py]: I ended up writing {notification**s**_count}, wouldn't that make sense to plural it wrt the other params ?

File: py3status/modules/github.py [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+
+GITHUB_API_URL = 'https://api.github.com'
+GITHUB_NOTIFICATION_URL = 'https://github.com/notifications'
+
+
+class Py3status:
+    auth_token = None
+    button_notifications = 1
+    cache_timeout = 60
```
Comment (ultrabug) [inline py3status/modules/github.py]: 300 sounds more reasonable no ?

**Comment by tobes** [comment]
Ok I've made changes.

* :bell: -> `N`

* `{repo}` short repo name `{repo_full}` repo full name

* `{notification_count}` -> `{notifications_count}`

I left the timeout at 60 as that feels responsive we are allowed 10 search requests per min without being logged in and get I think 30 if username/auth_token.  If you really want I can make 300 but for notifications that feels a bit slow (maybe that's just me)

**Comment by ultrabug** [comment]
Fine, thanks a lot !!

**Pull request merged**